src/target/riscv/riscv.c

```diff
@@ -2043,11 +2062,51 @@ int set_debug_reason(struct target *target, enum riscv_halt_reason halt_reason)
 	return ERROR_OK;
 }
 
+int sample_memory(struct target *target)
+{
+	RISCV_INFO(r);
+
+	if (!sample_buf.buf)
+		return ERROR_OK;
+
+    LOG_DEBUG("buf used/size: %d/%d", sample_buf.used, sample_buf.size);
+
+	uint64_t start = timeval_ms();
+	riscv_sample_buf_maybe_add_timestamp();
+	if (r->sample_memory) {
+		int result = r->sample_memory(target, &sample_buf, &sample_config,
+									  start + TARGET_DEFAULT_POLLING_INTERVAL);
```

Comment:
Should the duration of the sampling reflect the current (configurable) polling interval instead of always using the default?

Reply:
Where is the polling interval configurable? I don't see a command, and target.c defines
`static const int polling_interval = TARGET_DEFAULT_POLLING_INTERVAL;`

Reply:
You're right, sorry for this misinformation. I was under the impression it is configurable but it is not. I think there was a merge request about this in the upstream, which was apparently not merged in the end.